packages/shared/utils/is.ts

```diff
@@ -21,8 +21,13 @@ export const rand = (min: number, max: number) => {
 }
 export const hasOwn = <T extends object, K extends keyof T>(val: T, key: K): key is K => Object.prototype.hasOwnProperty.call(val, key)
 
+export const isIpad = /* #__PURE__ */ window?.navigator?.maxTouchPoints > 2 && /iPad|Macintosh/.test(window?.navigator.userAgent)
```

Comment:
I don't think we need to test with `maxTouchPoints`, any specific reason?

Reply:
The new iPad doesn't have `iPad` in the userAgent, so we can't detect if it's an iPad easily by matching userAgent, `maxTouchPoints` is the most frequently used in the community to solve this problem

Reply:
https://developer.apple.com/forums/thread/119186
Here are some descriptions of this problem